kolibri/plugins/coach/assets/src/app.js

```diff
@@ -22,7 +22,22 @@ const REPORTS_URL_PATTERN = [
   ':sort_column',
   ':sort_order',
 ].join('/');
-*/
+
+
+/* find the keys that differ between the old and new params */
+function _diffKeys(newParams, oldParams) {
+  if (!oldParams) {
+    return Object.keys(newParams);
+  }
+  const diffKeys = [];
+  Object.entries(newParams).forEach(([key, value]) => {
+    if (oldParams[key] !== value) {
+      diffKeys.push(key);
+    }
+  });
+  return diffKeys;
+}
+
```

Comment:
This can be written in terms of `filter`, e.g.:

```javascript
return Object.keys(newParams).filter((key) => newParams[key] !== oldParams[key])
```